x-pack/plugins/cases/server/client/cases/bulk_update.ts

```diff
@@ -344,15 +353,22 @@ export const bulkUpdate = async (
       return acc;
     }, new Map<string, CaseSavedObjectTransformed>());
 
-    const { nonExistingCases, conflictedCases, casesToAuthorize } = partitionPatchRequest(
-      casesMap,
-      query.cases
-    );
+    const { nonExistingCases, conflictedCases, casesToAuthorize, reopenedCases } =
+      partitionPatchRequest(casesMap, query.cases);
 
-    await authorization.ensureAuthorized({
-      entities: casesToAuthorize,
-      operation: Operations.updateCase,
-    });
+    const operationsToAuthorize =
+      reopenedCases.length > 0
+        ? [Operations.reopenCase, Operations.updateCase]
+        : [Operations.updateCase];
+
+    await Promise.all(
```

Comment:
A reminder of my comment about having the authorization class accept an array of operations  https://github.com/elastic/kibana/pull/194898#discussion_r1800888933.

Reply:
Yep, would be happy to pair on this or you can provide guidance on the files to update?

Reply:
On the `x-pack/plugins/cases/server/authorization/authorization.ts` there is the `Authorization` class of Cases. The changes would be:

```

  public async ensureAuthorized({
    entities,
    operations,
  }: {
    entities: OwnerEntity[];
    operations: OperationDetails[]; <--- an array of operations
  }) { await this._ensureAuthorized(uniqueOwners, operations); // rest of code omitted }
  

  private async _ensureAuthorized(owners: string[], operations: OperationDetails[]) {
    // rest of code

    if (securityAuth && this.shouldCheckAuthorization()) {
      const requiredPrivileges: string[] = operations.flatMap((operation) => owners.map((owner) =>
        securityAuth.actions.cases.get(owner, operation.name)
      )); <-- all possible pairs of operations and owners
      
      // rest of code
}
  

```

Reply:
@cnasikas In 8447c294123ba2a74db9312740464680b071674c I changed this function to support taking a single operation OR an array of operations. All existing tests/usages of it will work exactly as before, and we don't have to change 50+ different files to comply with the new signature. Added some tests for the array flavor of the function as well, let me know if this is ok instead.